Explorer/Assets/DCL/SceneLoadingScreens/LoadingScreen/ILoadingScreen.cs

```diff
@@ -8,25 +8,28 @@ namespace DCL.SceneLoadingScreens.LoadingScreen
 {
     public interface ILoadingScreen
     {
-        UniTask<Result> ShowWhileExecuteTaskAsync(Func<AsyncLoadProcessReport, CancellationToken, UniTask<Result>> operation,
-            CancellationToken ct);
+        UniTask<EnumResult<TaskError>> ShowWhileExecuteTaskAsync(
```

Comment:
Why the change from `Result` to `EnumResult`? According to the [Wiki](https://github.com/decentraland/unity-explorer/wiki/Architecture-Overview#exceptions-free-async-flow), the Result contract should be used in this flows.

Reply:
I haven't change Wiki yet, but TaskError gives more strict info of result's state (Cancelled, Timeout and etc)

Reply:
Aren't quite the same, but with an enum to limit them? They can even be described as `Result`, as implemented in the method. 

`
public Result AsResult()
{
    if (Success)
        return Result.SuccessResult();

    var error = Error!.Value;
    return Result.ErrorResult($"{error.State}: {error.Message}");
} 
`

We even have `Result<T>`, that is also quite similar

My concern is that we are creating a divergence on how we return an operation's result. Misha has the same concern as written at the end of this [post](https://github.com/decentraland/unity-explorer/pull/2913#issue-2702173216)

The Wiki was written less than a month ago so we could all adhere to a contract; if we want to change it, we should discuss it. 

Anyways, I don't want to hold this PR because of this. When we are all back, we can discuss getting rid of the `Result` contract and using `EnumResult` moving forward. After we have a decision, we would need a cleanup PR to adhere to it.

Reply:
Agreed, we should unify them. I found that Result<T> is limiting in cases of task implementations when it suits good for default methods without async